api/api.go

```diff
@@ -116,22 +168,23 @@ func (srv *Server) initAPI() {
 	// Wallet API Calls
 	if srv.wallet != nil {
 		router.GET("/wallet", srv.walletHandler)
-		router.POST("/wallet/033x", srv.wallet033xHandler)
-		router.GET("/wallet/address", srv.walletAddressHandler)
+		router.POST("/wallet/033x", requirePassword(srv.wallet033xHandler, password))
+		router.GET("/wallet/address", requirePassword(srv.walletAddressHandler, password))
 		router.GET("/wallet/addresses", srv.walletAddressesHandler)
-		router.GET("/wallet/backup", srv.walletBackupHandler)
-		router.POST("/wallet/init", srv.walletInitHandler)
-		router.POST("/wallet/lock", srv.walletLockHandler)
-		router.POST("/wallet/seed", srv.walletSeedHandler)
-		router.GET("/wallet/seeds", srv.walletSeedsHandler)
-		router.POST("/wallet/siacoins", srv.walletSiacoinsHandler)
-		router.POST("/wallet/siafunds", srv.walletSiafundsHandler)
-		router.POST("/wallet/siagkey", srv.walletSiagkeyHandler)
+		router.GET("/wallet/backup", requirePassword(srv.walletBackupHandler, password))
+		router.POST("/wallet/init", requirePassword(srv.walletInitHandler, password))
+		router.POST("/wallet/lock", requirePassword(srv.walletLockHandler, password))
+		router.POST("/wallet/seed", requirePassword(srv.walletSeedHandler, password))
+		router.GET("/wallet/seeds", requirePassword(srv.walletSeedsHandler, password))
+		router.POST("/wallet/siacoins", requirePassword(srv.walletSiacoinsHandler, password))
+		router.POST("/wallet/siafunds", requirePassword(srv.walletSiafundsHandler, password))
+		router.POST("/wallet/siagkey", requirePassword(srv.walletSiagkeyHandler, password))
```

Comment:
Is it that significant if we wrap every route? It would definitely improve privacy surface area.

I don't think it's that bad to have this level of customization, though it does require you to think about passwords every time you make a change. It adds complexity to the codebase and should be considered carefully. If we choose to expose endpoints, it's not really something that we can reverse easily in the future. But the other way is easier, if you decide that something doesn't need to be secret, you can still accept a password on the call.